### ISSUE TITLE
Skips shortcut icons

### ISSUE THREAD

**Issue by Quezler** [mat/besticon#25]

**Comment by Quezler** [comment]
For a project of mine i display an overview of domains/sites with associated favicon through here, and i have noticed that sometimes a site with a **shortcut icon** tag does not show up in the icon list.

A few example sites not getting picked up:

[www.heartforgardens.nl](https://icons.better-idea.org/icon?url=www.heartforgardens.nl&size=16..64..500)

**Comment by Quezler** [comment]
(tagging @mat manually since the primary description is empty, so any notification would not relay the new content)

**Comment by mat** [comment]
Hi @Quezler, thanks for your report! I did some investigation and the problem here is not the shortcut icon tag itself but the **image formats** used for those icons, namely:

- www.heartforgardens.nl uses jpeg
- www.samson.nl uses jpeg
- www.vanginkeltuinen.nl uses bmp

Right now only png, gif and ico are supported (see readme). I removed support for jpeg and bmp a while ago in f8bd3e3 because of some problems I fail to remember.

I am willing to try adding back support for `jpg` and `bmp` - but we should not change the defaults for the `formats` parameter and instead stay with `png,ico,gif` so we don't break existing clients which might not support bmp and jpeg.

Would you like to give this a try?

**Comment by Quezler** [comment]
Well since this tool simply redirects to the url it finds, I don't think it would be to much of a problem if we somehow manage to add like **&anyformat=true** or something along those lines?

**Comment by mat** [comment]
Yeah, I've thought about this in the past when I decided to use an explicit format parameter as this can be a real problem:

- iOS and Android don't have support for the ancient `ico natively
- should we support SVG one day (#18) this would likely also be a problem on all non-web platforms. (same would be true for webp format e.g.)
